apps/angular/projection/src/app/component/city-card/city-card.component.ts

```diff
@@ -1,13 +1,40 @@
+import { AsyncPipe } from '@angular/common';
 import { Component, OnInit } from '@angular/core';
+import { CityStore } from '../../data-access/city.store';
+import { randomCity } from '../../data-access/fake-http.service';
+import { City } from '../../model/city.model';
+import { DataCardComponentBase, DataStoreBase } from '../../shared';
+import { CardComponent } from '../../ui/card/card.component';
 
 @Component({
   selector: 'app-city-card',
-  template: 'TODO City',
+  template: `
+    <app-card
+      [list]="data$ | async"
+      (addNewItem)="onAddNewItem()"
```

Comment:
Instead of `onAddNewItem` you could be more specific about the handler → `addNewCitiy()`.

`#personalPerference` `#no-blocker`

Reply:
Yeah, as you noticed later the intention was to move all repetitive code out of components, to make it possible to handle it in one place.